### ISSUE TITLE
FactorVAE encoder graph detaching during discriminator loss optimization

### ISSUE THREAD

**Issue by willtop** [YannDubs/disentangling-vae#77]
In FactorVAE, for the discriminator optimization, we don't want the gradients incurred in the VAE to update the VAE parameters, and thus the detach operation here:
https://github.com/YannDubs/disentangling-vae/blob/f0452191bab6d94eba0b4e6a065f74dcfd54ac52/disvae/models/losses.py#L287

However, wouldn't we need to detach the first set of latent vectors as well (i.e. latent_sample1), now that we have to move the optimizer.step() at the end (due to the in-place modification error already addressed as one closed issue in the repo)?

I ran a debugging session and indeed gradient changes are observed on the VAE encoder from d_tc_loss.backward().
